### DISCUSSION TITLE
Is there a way to run some logic before a put()?

### DISCUSSION THREAD

**Discussion by Kazade** [isar/isar#1648] [Help]
I'd like a generic way to run some code when I insert an object into a collection, for example, run some cleanup or validation, perform some logging etc.

I come from a Django background, and there I can just override the "save" method on the model to do this - is there a way I can achieve something similar in Isar?

**Comment by Tr736** [comment]
couldn't you do something like this....

```dart   /// addItem
  Future<void> addItem(Item item) async {
   try {
  // logging or validation
if (failedValidation)  throw Exception('some error')
    isar.writeTxnSync(() {
      store.putSync(item);
    });
  } catch (e,s) {
// some error handling
}
  }`

**Reply by Kazade** [reply]
I mean I can, of course, but I was wondering if there's some-way I can hook before put() more generically. Isar has watchers for *after* insert, but not before. 

With the above example all it takes is for someone to forget to call addItem instead of put and it'll cause a bug.
